integration_tests/annotations/annotations_test.go

```diff
@@ -0,0 +1,191 @@
+}
+
+type EmbeddedProto2 struct {
+	*proto2apb.Annotation
+}
+
+func (e *EmbeddedProto2) MarshalJSON() ([]byte, error) {
+	return protojson.Marshal(e)
+}
+
+func (e *EmbeddedProto2) UnmarshalJSON(b []byte) error {
+	return protojson.Unmarshal(b, e)
+}
+
+func NewEmbeddedProto2(s string) *EmbeddedProto2 {
+	return &EmbeddedProto2{
+		Annotation: &proto2apb.Annotation{Comment: proto.String(s)},
+	}
+}
+
+func TestProtoAnnotation(t *testing.T) {
+	tests := []struct {
+		desc                      string
+		inAnnotation              ygot.Annotation
+		wantJSON                  string
+		wantMarshalErrSubstring   string
+		wantUnmarshalErrSubstring string
+	}{{
+		desc:         "included proto3 annotation",
+		inAnnotation: &IncludedProto3{a: &apb.Annotation{Comment: "hello"}},
+		wantJSON: `
+		{
+			"@": [
+			   {
+				  "comment": "hello"
+			   }
+			]
+		 }`,
+	}, {
+		desc:         "embedded proto3 annotation",
+		inAnnotation: NewEmbeddedProto3("hello world"),
+		wantJSON: `
+		{
+			"@": [
+			   {
+				  "comment": "hello world"
+			   }
+			]
+		 }`,
+	}, {
+		desc:         "included proto2 annotation",
+		inAnnotation: &IncludedProto2{a: &proto2apb.Annotation{Comment: proto.String("hello")}},
+		wantJSON: `
+		{
+			"@": [
+			   {
+				  "comment": "hello"
+			   }
+			]
+		 }`,
+	}, {
+		desc:         "embedded proto2 annotation",
+		inAnnotation: NewEmbeddedProto2("hello world"),
+		wantJSON: `
+		{
+			"@": [
+			   {
+				  "comment": "hello world"
+			   }
+			]
+		 }
+		`,
+	}}
+
+	isEmptyDiff := func(d *gpb.Notification) bool {
+		return len(d.Update) == 0 && len(d.Delete) == 0
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			d := &oc.Device{}
+			d.ΛMetadata = append(d.ΛMetadata, tt.inAnnotation)
+
+			got, err := ygot.EmitJSON(d, &ygot.EmitJSONConfig{
+				Format: ygot.RFC7951,
+			})
+			if diff := errdiff.Substring(err, tt.wantMarshalErrSubstring); diff != "" {
+				t.Fatalf("did not get expected marshalling error, %s", diff)
+			}
+
+			wantJ := map[string]interface{}{}
+			if err := json.Unmarshal([]byte(tt.wantJSON), &wantJ); err != nil {
+				t.Fatalf("cannot unmarshal expected JSON, err: %v", err)
+			}
+			gotJ := map[string]interface{}{}
+			if err := json.Unmarshal([]byte(got), &gotJ); err != nil {
+				t.Fatalf("cannot unmarshal received JSON, err: %v", err)
+			}
+
+			if diff := cmp.Diff(gotJ, wantJ); diff != "" {
+				t.Fatalf("did not get expected marshalled JSON, diff(-got,+want):\n%s", diff)
+			}
+
+			nd := &oc.Device{}
+			err = oc.Unmarshal([]byte(got), nd)
+			if diff := errdiff.Substring(err, tt.wantUnmarshalErrSubstring); diff != "" {
+				t.Fatalf("did not get expected unmarshalling error, %s", diff)
+			}
+
+			diff, err := ygot.Diff(nd, d)
+			if err != nil {
+				t.Fatalf("error diffing expected and received unmarshalled content, %v", err)
+			}
+			if !isEmptyDiff(diff) {
+				t.Fatalf("did not get expected unmarshalled output, not equal to input! Diff: %s", diff)
```

Comment:
nit: extra "!" in comment

Reply:
Fixed - amongst the many github-actions CI issues below :-(